typespec-extension/src/code-model-builder.ts

```diff
@@ -1923,7 +1923,9 @@ export class CodeModelBuilder {
     });
 
     // cache this now before we accidentally recurse on this type.
-    this.schemaCache.set(type, dictSchema);
+    if (!this.schemaCache.has(type)) {
```

Comment:
why do we add the check? Do we meet any issue?

Reply:
The add to Record is for this issue.

But generally we should not overwrite the cache, anyway. Hence here too.

Reply:
At this line, we call intentionally call `processDictionarySchema` to avoid the cache (as we want to get a new DictonarySchema).
https://github.com/Azure/autorest.java/blob/9af51f7a1ab414f79c267946c1da6e11dfc73d01/typespec-extension/src/code-model-builder.ts#L2240

However, this line still write cache. Hence the problem.

Reply:
ok, maybe ideally we can let `schemaCache` do this check, instead of having to check every time when we set cache? (I'm ok with having the check here first)

Reply:
It already handles it https://github.com/Azure/autorest.java/blob/9af51f7a1ab414f79c267946c1da6e11dfc73d01/typespec-extension/src/type-utils.ts#L45-L49

Here just 2 special process.